test/integration/oauth_token_endpoint_test.go

```diff
@@ -0,0 +1,158 @@
+		MappingMethod:   "claim",
+		Provider:        &configapi.AllowAllPasswordIdentityProvider{},
+	}
+	clusterAdminKubeConfig, err := testserver.StartConfiguredMaster(masterOptions)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	clientConfig, err := testutil.GetClusterAdminClientConfig(clusterAdminKubeConfig)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	transport := knet.SetTransportDefaults(&http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	})
+
+	// Hit the token request endpoint
+	masterURL, err := url.Parse(clientConfig.Host)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	masterURL.Path = "/oauth/token/request"
+	tokenReq, err := http.NewRequest("GET", masterURL.String(), nil)
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+	tokenResp, err := transport.RoundTrip(tokenReq)
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+	defer tokenResp.Body.Close()
+
+	if tokenResp.StatusCode != http.StatusFound {
+		t.Fatalf("unexpected response code %v", tokenResp.StatusCode)
+	}
+
+	// Follow location header to auth
+	authRedirect, err := url.Parse(tokenResp.Header.Get("Location"))
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+
+	authReq, err := http.NewRequest("GET", authRedirect.String(), nil)
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+
+	authReq.SetBasicAuth("foo", "bar")
+	authReq.Header.Set("Accept", "text/html; charset=UTF-8")
+	authResp, err := transport.RoundTrip(authReq)
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+	defer authResp.Body.Close()
+
+	if authResp.StatusCode != http.StatusFound {
+		t.Fatalf("unexpected response code %v", authResp.StatusCode)
+	}
+
+	// Follow location header to token display
+	displayReq, err := http.NewRequest("GET", authResp.Header.Get("Location"), nil)
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+
+	displayReq.Header.Set("Accept", "text/html; charset=UTF-8")
+	displayResp, err := transport.RoundTrip(displayReq)
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+	defer displayResp.Body.Close()
+
+	if displayResp.StatusCode != http.StatusOK {
+		t.Fatalf("unexpected response code %v", displayResp.StatusCode)
+	}
+
+	// Parse the HTML body for the API token contained in a <code></code> block
+	tokenizer := html.NewTokenizer(displayResp.Body)
+
+	var apiToken string
+	for {
+		token := tokenizer.Next()
+		if token == html.ErrorToken {
+			break
+		}
+		if token != html.StartTagToken {
+			continue
+		}
+		td := tokenizer.Token()
+		if td.Data != "code" {
+			continue
+		}
+
+		// The code tag is followed by the apiToken text
+		token = tokenizer.Next()
+		if token == html.TextToken {
+			tt := tokenizer.Token()
+			apiToken = tt.Data
+			break
+		}
+	}
+
+	if len(apiToken) == 0 {
+		t.Fatalf("no API token found in HTML body")
+	}
+
+	// Check the use of the bearer token.
+	masterURL.Path = "/oapi/v1/users/~"
```

Comment:
I think once you have the token you should be able to build a client that uses it.  Then check to make sure the `whoami` call does not error and the user that it points to is correct.